File: NeetCode/Arrays & Hashing/Contains Duplicate/ContainsDuplicate.js
const containsDuplicate = (nums) => {
  const numsHashset = new Map();

  for (let i = 0; i < nums.length; i++) {
    if (numsHashset.has(nums[i])) {
      return true;
    }
    numsHashset.set(nums[i], true);
  }
  return false;
};

console.log(containsDuplicate([2, 3, 4, 1]));

File: NeetCode/Arrays & Hashing/Contains Duplicate/ContainsDuplicate.py
# Time O(n)
# Space O(n)

# https://leetcode.com/problems/contains-duplicate/description/

class Solution:
    def containsDuplicate(self, nums):
        hashset = set()

        for n in nums:
            if n in hashset:
                return True
            hashset.add(n)
        return False

result = Solution()
print(result.containsDuplicate([1,3,5,2]))
